### PR TITLE
Add availability to set environment variables for instance service

### LINKED ISSUE #331: [4pt] Allow to customize env variable for services
It would be great to have ability to specify env variables per instance. We need it to enable some features on subset of instances.

Example. I want to specify "ENABLE_1_MILLON_RPS=1" and then check it via os.getenv.
```lua
if os.getenv('ENABLE_1_MILLON_RPS') ~= nil then
    box.cfg{}
else
    require('ffi').new('char*')[0]
end

```

### PR DESCRIPTION
**Pull request by hackallcode**
Closes #331

Before the patch, it was impossible to set environment variables for services,
except for some that are used in the unit file template in this role
(so, it was possible only for TGZ).

Now it's possible to set environment variables for all instace services for any type of package.

### CODE REVIEW

File: tasks/steps/configure_instance.yml [hunk 1]
```diff
@@ -3,6 +3,10 @@
 - when: not expelled
   tags: cartridge-instances
   block:
+    - name: 'BLOCK: Extend systemd unit files'
```
Comment (olegrok) [inline tasks/steps/configure_instance.yml:6]: Why is `BLOCK` in uppercase?

File: tasks/steps/blocks/extend_systemd_unit_files.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    path: '{{ (cartridge_systemd_dir, instance_info.systemd_service_dir) | cartridge_path_join }}'
+  register: systemd_service_dir_res
+
+- name: 'Extend systemd unit files with environment'
+  copy:
+    dest: '{{ (cartridge_systemd_dir, instance_info.systemd_service_env_file) | cartridge_path_join }}'
+    content: |
+      [Service]
+      {% for key, value in cartridge_extra_env.items() %}
+      Environment={{ key }}={{ value }}
```
Comment (olegrok) [inline tasks/steps/blocks/extend_systemd_unit_files.yml]: Does it work fine if option contains spaces or some special characters? Don't you need some escaping?

File: tasks/steps/configure_instance.yml (hunk 1 above)
Comment (hackallcode) [reply tasks/steps/configure_instance.yml:6]: It's a style of role: to show that this task doesn't make any actions, it's start of block

Comment (olegrok) [reply tasks/steps/configure_instance.yml:6]: Ok. Thanks for explanation.

File: tasks/steps/blocks/extend_systemd_unit_files.yml (hunk 2 above)
Comment (hackallcode) [reply tasks/steps/blocks/extend_systemd_unit_files.yml]: No, but i fixed it and added in tests:
https://github.com/tarantool/ansible-cartridge/blob/3083d473becf787581795f26fa72c8fa057f8dc9/molecule/custom_env/converge.yml#L55-L58

**Review by olegrok — DISMISSED** [review]

File: defaults/main.yml [hunk 3]
```diff
@@ -99,6 +99,7 @@ cartridge_defaults: {}
 config: null
 cartridge_runtime_params: null
 zone: null
+cartridge_extra_env: null
```
Comment (dokshina) [inline defaults/main.yml:102]: `cartridge_instance_env`?

File: tasks/steps/blocks/extend_systemd_unit_files.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+---
+
+- name: 'Create directory for systemd unit files extensions'
+  file:
+    state: '{{ "directory" if cartridge_extra_env else "absent" }}'
+    recurse: '{{ true if cartridge_extra_env else false }}'
+    path: '{{ (cartridge_systemd_dir, instance_info.systemd_service_dir) | cartridge_path_join }}'
```
Comment (dokshina) [inline tasks/steps/blocks/extend_systemd_unit_files.yml]: It's `systemd_service_dir_name` if you need to join it. BTW, why don't you compute abspath on getting instance info?

File: tasks/steps/blocks/extend_systemd_unit_files.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  register: systemd_service_dir_res
+
+- name: 'Extend systemd unit files with environment'
+  copy:
+    dest: '{{ (cartridge_systemd_dir, instance_info.systemd_service_env_file) | cartridge_path_join }}'
```
Comment (dokshina) [inline tasks/steps/blocks/extend_systemd_unit_files.yml]: The same as above.

File: doc/variables.md [hunk 6]
```diff
@@ -87,6 +87,8 @@ For more details see [scenario documentation](/doc/scenario.md).
 * `config` (`dict`, required): [instance configuration](/doc/instances.md);
 * `zone` (`string`): instance zone (available since
   [Cartridge 2.4.0](https://github.com/tarantool/cartridge/releases/tag/2.4.0));
+* `cartridge_extra_env` (`dict`): environment variables for instance service
```
Comment (dokshina) [inline doc/variables.md:90]: What about describing the difference between `null` and `{}`?

File: defaults/main.yml (hunk 3 above)
Comment (hackallcode) [reply defaults/main.yml:102]: You can use this variable for all instances

File: doc/variables.md (hunk 6 above)
Comment (hackallcode) [reply doc/variables.md:90]: Already....

**Review by dokshina — DISMISSED** [review]

File: molecule/custom_env/converge.yml [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+    - name: 'Get info about service directory'
+      stat:
+        path: '{{ instance_info.systemd_service_dir }}'
+      register: stat_res
+
+    - name: 'Check service directory'
+      assert:
+        fail_msg: "Directory should exist"
+        success_msg: "Directory shouldn't exist"
```
Comment (dokshina) [inline molecule/custom_env/converge.yml:132]:
```
        fail_msg: "Directory exist"
        success_msg: "Directory shouldn't exist"
```

File: molecule/custom_env/converge.yml [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+        fail_msg: 'Received bad environment'
+        success_msg: 'Received environment is OK'
+        that: eval_res == []
+
+    - name: 'Get info about service directory'
+      stat:
+        path: '{{ instance_info.systemd_service_dir }}'
+      register: stat_res
+
+    - name: 'Check service directory'
```
Comment (dokshina) [inline molecule/custom_env/converge.yml]: `Check service directory doesn't exist`

File: molecule/custom_env/converge.yml [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+    - name: 'Get info about service environment file'
+      stat:
+        path: '{{ instance_info.systemd_service_env_file }}'
+      register: stat_res
+
+    - name: 'Check service environment file'
+      assert:
+        fail_msg: "Environment file shouldn't exist"
+        success_msg: "Environment file should exist"
```
Comment (dokshina) [inline molecule/custom_env/converge.yml:90]:
```
        fail_msg: "Environment file doesn't exist"
        success_msg: "Environment file should exist"
```

File: molecule/custom_env/converge.yml [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+    - name: 'Get info about service directory'
+      stat:
+        path: '{{ instance_info.systemd_service_dir }}'
+      register: stat_res
+
+    - name: 'Check service directory'
+      assert:
+        fail_msg: "Directory should exist"
+        success_msg: "Directory shouldn't exist"
```
Comment (dokshina) [inline molecule/custom_env/converge.yml:41]:
```
        fail_msg: "Directory exists"
        success_msg: "Directory shouldn't exist"
```

**Review by dokshina — COMMENTED** [review]
Maybe I've missed other sich places, please check

**Review by dokshina — APPROVED** [review]

**Pull request merged**